File: Backend/CookItUp/src/main/kotlin/com/sorsix/cookitup/model/dto/RecipeDTO.kt
package com.sorsix.cookitup.model.dto

import lombok.Data
import org.springframework.web.multipart.MultipartFile

@Data
data class RecipeDTO(
    val name:String,
    val description:String,
    val numPersons:Int,
    val difficultyLevel: String,
    val prepTime:Int,
    val categoryList: List<String>,
    val ingredientList: List<IngredientIsInRecipeDTO>,
)

File: Frontend/src/app/models/recipedto.ts
import { Category } from './category';
import { DifficultyLevel } from './difficultyLevel';
import { IngInRecipe } from './ingInRecipe';
import { Ingredient } from './ingredient';

export class RecipeDTO {
  name!: string;
  description!: string;
  numPersons!: number;
  difficultyLevel!: DifficultyLevel;
  prepTime!: number;
  categoryList!: Category[];
  ingredientList!: IngInRecipe[];
}
